src/background/services/balances/BalanceAggregatorService.ts

```diff
@@ -143,14 +144,15 @@ export class BalanceAggregatorService implements OnLock, OnUnlock {
       for (const [chainId, chainBalances] of freshData) {
         for (const [address, addressBalance] of Object.entries(chainBalances)) {
           aggregatedBalances[chainId] = {
+            ...aggregatedBalances[chainId],
```

Comment:
This fixes a bug which was causing existing balances for the address to not be kept in cache when updating with partial values.

Reply:
I'll take another look at it 👀

Reply:
Yeah, we needed something like this instead: 24f970b16